pygmt/exceptions.py

```diff
@@ -130,3 +130,48 @@ def __init__(self, dtype: object, /, reason: str | None = None):
         if reason:
             msg += f" {reason}"
         super().__init__(msg)
+
+
+class GMTParameterError(GMTError):
+    """
+    Raised when parameters are missing or invalid.
+
+    Parameters
+    ----------
+    required
+        Names of required parameters.
+    require_any
+        Names of parameters where at least one must be specified.
+    exclusive
+        Names of mutually exclusive parameters.
+    reason
+        Detailed reason why the parameters are invalid.
+    """
+
+    def __init__(
+        self,
+        *,
+        required: Set[str] | None = None,
+        require_any: Set[str] | None = None,
+        exclusive: Set[str] | None = None,
+        reason: str | None = None,
+    ):
+        msg = ""
+        if required:
+            msg = (
+                "Required parameter(s) are missing: "
+                f"{', '.join(repr(par) for par in required)}."
+            )
```

Comment:
For this case where the parameter(s) is required/compulsory, I'm wondering if we can move towards having the error being thrown natively by Python, instead of writing code like:

```python
if kwargs.get("F") is None:
    raise GMTParameterError(required={"filter_type"})
```

Should we revisit #2896 which mentions [PEP692](https://peps.python.org/pep-0692/), and also think about using [PEP655](https://peps.python.org/pep-0655/)'s `typing.Required` qualifier (https://typing.python.org/en/latest/spec/callables.html#required-and-non-required-keys)? Maybe a step towards #262 is to disallow single character arguments for required parameters, so that we can have native Python errors?

Reply:
The main issue is that, with the new alias system #4000 implemented, `**kwargs` will no longer be needed. We will still keep `**kwargs` to support single-letter option flags, but other parameters won't be passed via `kwargs` anymore.